beamer/tests/agent/test_request.py

```diff
@@ -147,6 +147,47 @@ def test_fill_and_claim(request_manager, token, agent, allow_unlisted_pairs):
     assert claim.claimer == agent.address
 
 
+@pytest.mark.parametrize("allowance", [None, "-1", "1", "3"])
+def test_allowance(request_manager, fill_manager, token, agent, allowance):
+    """
+    Allowance can have 4 different values
+
+    If None (backwards compatibility), the requested amount will be approved.
+    If -1, approved value will be type(uint256).max
+    If allowance < requested amount, agent will not fill
+    If allowance >= requested amount, configured value will be used
+    """
+    requester, target = alloc_accounts(2)
+    amount = 2
+    request_id = make_request(request_manager, token, requester, target, amount)
+    if allowance is None:
+        actual_allowance = amount
+    elif allowance == "-1":
+        actual_allowance = 2**256 - 1
+    else:
+        actual_allowance = int(allowance)
+
+    try:
+        with Sleeper(5) as sleeper:
+            while (
+                request := agent.context.requests.get(request_id)
+            ) is None or request.filler != agent.address:
+                sleeper.sleep(0.1)
+    except Timeout:
+        # Agent did not fill
+        assert actual_allowance < amount
+
+    else:
+        allowance_after = token.allowance(agent.address, fill_manager)
+
+        # Some token contracts implement either an unlimited allowances at type(uint256).max
+        # other start decreasing the allowance from the highest amount.
+        if actual_allowance == 2**256 - 1:
+            assert allowance_after in [actual_allowance - amount, actual_allowance]
+        else:
+            assert token.allowance(agent.address, fill_manager) == actual_allowance - amount
```

Comment:
Can we just use `allowance_after` here instead of making another call to `allowance`?